homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,213 @@
+            )
+        )
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self, **kwargs):
+        """Update the sensor."""
+
+        departure_time = utcnow() + timedelta(
+            minutes=self.config_entry.options.get("offset", 0)
+        )
+
+        payload = {
+            "station": self.config_entry.data["station"],
+            "time": {
+                "date": departure_time.strftime("%d.%m.%Y"),
+                "time": departure_time.strftime("%H:%M"),
+            },
+            "maxList": MAX_LIST,
+            "maxTimeOffset": MAX_TIME_OFFSET,
+            "useRealtime": self.config_entry.options.get("realtime", False),
+        }
+
+        if "filter" in self.config_entry.options:
+            payload.update({"filter": self.config_entry.options["filter"]})
+
+        try:
+            data = await self.gti.departureList(payload)
+        except InvalidAuth as error:
+            if self._last_error != InvalidAuth:
+                _LOGGER.error("Authentication failed: %r", error)
+                self._last_error = InvalidAuth
+            self._available = False
+        except ClientConnectorError as error:
+            if self._last_error != ClientConnectorError:
+                _LOGGER.warning("Network unavailable: %r", error)
+                self._last_error = ClientConnectorError
+            self._available = False
+        except Exception as error:  # pylint: disable=broad-except
+            if self._last_error != error:
+                _LOGGER.error("Error occurred while fetching data: %r", error)
+                self._last_error = error
+            self._available = False
+
+        if data["returnCode"] == "OK" and data.get("departures"):
+
+            if self._last_error == ClientConnectorError:
+                _LOGGER.debug("Network available again")
+
+            self._last_error = None
+
+            departure = data["departures"][0]
+            self._available = True
+            self._state = (
+                departure_time
+                + timedelta(minutes=departure["timeOffset"])
+                + timedelta(seconds=departure.get("delay", 0))
+            )
+
+            self.attr[ATTR_LINE] = departure["line"]["name"]
+            self.attr[ATTR_ORIGIN] = departure["line"]["origin"]
+            self.attr[ATTR_DIRECTION] = departure["line"]["direction"]
+            self.attr[ATTR_TYPE] = departure["line"]["type"]["shortInfo"]
+            self.attr[ATTR_ID] = departure["line"]["id"]
+            self.attr[ATTR_DELAY] = departure.get("delay", 0)
+
+            departures = []
+            for departure in data["departures"]:
+                departures.append(
+                    {
+                        ATTR_DEPARTURE: departure_time
+                        + timedelta(minutes=departure["timeOffset"])
+                        + timedelta(seconds=departure.get("delay", 0)),
+                        ATTR_LINE: departure["line"]["name"],
```

Comment:
Please create a local variable before appending the departure e.g. for `departure["line"]` since it is used so much here.

Reply:
just out of interest, is this change for ease of reading or are there performance boosts by using a local variable?

Reply:
Mostly for ease of reading, not sure how much the performance is improved by this change.